### ISSUE TITLE
Add a way to build a custom project

### ISSUE THREAD

**Issue by nacho** [wingtk/gvsbuild#306]
@guruDanny67 I think it would be great to have a param --custom myproject.project or something similar so like this I could add a custom project to build from git to add continuous integration for glib-networking using gvsbuild.

**Comment by nacho** [comment]
indeed, that's what I thought yes.

**Comment by guruDanny67** [comment]
Adding one (or more) python scripts to the startup should be no big pain: to see if it works, before making big changes, we can import all the scripts found under a directory (user_scripts, with a lot of fantasy): if there is something (*.py) under that we import it after the import of tools, projects & group

**Comment by guruDanny67** [comment]
what about an option --user-scripts-dir dir1,dir2 that loads all the scripts found in the specified directories?

**Comment by nacho** [comment]
That's fine for me too.

**Comment by guruDanny67** [comment]
Just put a branch (wip/user-prj) with a simple implementation & a simple example script (that rebuilds the zlib, using another name).

What comes to mind that should be done to really use the system is:
 * Option, on the single project, to override the one in the base scripts (now it gives you an error, porject already defined)
 * The patches dir for the single project should be under the user dir: if I have the project 'gtk4' under gtk4-test the dir, for only that project, should be gtk4-test\gtk4. In this way, we can have full control on the generated project.

**Comment by nacho** [comment]
I guess we can go step by step, maybe make a pull request to review and then we can iterate if needed. Maybe for the normal use case it would already work

**Comment by guruDanny67** [comment]
Agree, I try to put at least the patches dir and then I will make the PR.
